frontend/src/components/conflictResolution/conflictResolutionItem.tsx

```diff
@@ -0,0 +1,182 @@
+import { IStudent } from "../../api/entities/StudentEntity";
+import { Form, Formik } from "formik";
+import useSWR, { useSWRConfig } from "swr";
+import {
+    getAllAssignmentsFromPage,
+    getValidAssignmentsUrlForStudent,
+    invalidateAssignment,
+} from "../../api/calls/AssignmentCalls";
+import { useEffect, useState } from "react";
+import ConflictResolutionRadioButton from "./conflictResolutionRadioButton";
+import { IAssignment } from "../../api/entities/AssignmentEntity";
+import { getProjectSkillOnUrl } from "../../api/calls/projectSkillCalls";
+import { capitalize } from "../../utility/stringUtil";
+import useTranslation from "next-translate/useTranslation";
+
+/**
+ * Properties needed by [ProjectSkillRegister].
+ */
+export interface ProjectSkillRegisterProps {
+    registerAssignment: (projectSkillUrl: string, assigmentUrl: string) => void;
+    removeAssignment: (assignmentUrl: string) => void;
+    assignment: IAssignment;
+}
+
+/**
+ * Component registering assignments and the projectsSkill they belong to. Needed as a workaround for rules of hooks.
+ * @param registerAssignment callBack registering so the component can register itself.
+ * @param removeAssignment callback removing the current assignment.
+ * @param assignment the assignment the hook handles.
+ */
+export function ProjectSkillRegister({
+    registerAssignment,
+    removeAssignment,
+    assignment,
+}: ProjectSkillRegisterProps) {
+    const { data: receivedSkill, error: skillError } = useSWR(
+        assignment._links.projectSkill.href,
+        getProjectSkillOnUrl
+    );
+
+    const receivedProjectSkillUrl = receivedSkill?._links.self.href;
+    const assignmentUrl = assignment._links.self.href;
+    const hasErrored = !!skillError;
+
+    useEffect(() => {
+        if (receivedProjectSkillUrl) {
+            registerAssignment(receivedProjectSkillUrl, assignmentUrl);
+        } else if (hasErrored) {
+            // Let us hope this doesn't happen too often
+            removeAssignment(assignmentUrl);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Should this be disabled?

Reply:
Yeah, the hooks are dependencies are configured correctly so this should be allowed.